src/HasEnums.php

```diff
@@ -132,6 +132,17 @@ protected function setEnumAttribute(string $key, $value)
             return $this;
         }
 
+        if ($this->isArrayOfEnums($key)) {
+            if (! is_array($value)) {
+                $modelClass = static::class;
+                throw new InvalidArgumentException("{$modelClass} [{$key}] expects array of enum values.");
```

Comment:
In these cases I'd prefer to use a dedicated exception class, what do you think?

Reply:
Yeah. Good point. Will add one also for the nullable exception.